### PR TITLE
support cross-compilation by not using the cfg macro in build.rs

### LINKED ISSUE #14: Cannot cross-compile pqcrypto-kyber
When trying to compile a project that depends on pqcrypto-kyber to another architecture, like so:
```
cargo build --target armv7-unknown-linux-gnueabihf
```

I get the following warnings:
```
  cargo:warning=armv7l-unknown-linux-gnueabihf-gcc: error: unrecognized command line option ‘-mavx2’
  cargo:warning=armv7l-unknown-linux-gnueabihf-gcc: error: unrecognized command line option ‘-mbmi2’
  cargo:warning=armv7l-unknown-linux-gnueabihf-gcc: error: unrecognized command line option ‘-mbmi’
  cargo:warning=armv7l-unknown-linux-gnueabihf-gcc: error: unrecognized command line option ‘-maes’
  cargo:warning=armv7l-unknown-linux-gnueabihf-gcc: error: unrecognized command line option ‘-mpopcnt’
```
after which the compilation fails.

I'm using version 0.6.5.

I'm not sure exactly why this happens, as for it to happen, [this](https://github.com/rustpq/pqcrypto/blob/master/pqcrypto-kyber/build.rs#L54-L59) should be executed, but this is behind a target_arch = "x86_64" #cfg.

### PR DESCRIPTION
**Pull request by noonien**
Fixes #14

### CODE REVIEW

File: implementations.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ traits_version: 0.3.2
 kems:
   kyber:
     version: 0.6.5
-    avx2_guard: 'cfg(all(not(disable_avx2), not(target_os = "windows"), not(target_os = "macos"), target_arch = "x86_64"))'
+    avx2_guard: '!is_windows && !is_macos && target_arch == "x86_64"'
```
Comment (thomwiggers) [inline implementations.yaml:9]: This is also not good enough: if you have an old CPU this will lead to illegal instructions.

**Review by thomwiggers — CHANGES_REQUESTED** [review]

File: implementations.yaml (hunk 1 above)
Comment (thomwiggers) [reply implementations.yaml:9]: Actually I should first have coffee before deciding such things — the above isn't true.

**Review by thomwiggers — CHANGES_REQUESTED** [review]
On a second look, this actually seems to work.

The only issue remaining seems to change ``DEFAULT_AVX2_GUARD`` in ``generate_implementations.py``.

**Comment by noonien** [comment]
Opps, missed that.

It's fixed now. Also did a `cargo check` and `cargo test ` in the workspace to make sure everything compiles, and runs.

There is a SIGSEGV in `pqcrypto-classicmceliece`, however, the error exists in the old code as well.

**Review by thomwiggers — CHANGES_REQUESTED** [review]
Sorry, I forgot to also ask that you do not remove the `disable_avx2` opt-in feature

**Comment by thomwiggers** [comment]
> There is a SIGSEGV in pqcrypto-classicmceliece, however, the error exists in the old code as well.

Yeah, I think there's some issue in the upstream code but I had a hard time locking it down.

**Comment by noonien** [comment]
I'm not exactly sure what the disable_avx opt-in feature is. Will it be passed using a cfg_attr macro? Because it's not defined in Config.tom. If so, are you sure that is the right place to use it like this? I'm not sure those config attrs will be available in build.rs.

**Comment by thomwiggers** [comment]
Some good questions :sweat_smile:  I think I used it via `-C disable_avx2` for testing.

**Comment by noonien** [comment]
Was that set in RUSTFLAGS?

**Comment by thomwiggers** [comment]
I think so, yeah

**Comment by noonien** [comment]
[RUSTFLAGS also doesn't work when cross-compiling](https://github.com/rust-lang/cargo/issues/4423). I'll add a feature to Cargo.toml.

**Comment by thomwiggers** [comment]
It's still helpful to have, it allows me to test the non-AVX2 codepaths on an AVX2 machine.

**Comment by noonien** [comment]
Fixed.

I added a feature in Cargo.toml for packages that support avx2. To run without avx2 support use:
```
cargo build --no-default-features
```

**Review by thomwiggers — APPROVED** [review]

**Comment by noonien** [comment]
Thanks, cheers!

:handshake:

**Comment by thomwiggers** [comment]
Thanks for the contribution

**Comment by noonien** [comment]
Any chance to get this change in a release?

**Comment by thomwiggers** [comment]
I've run the script.

I also should update everything with PQClean changes, but I'm not sure if I'll have time for that anytime soon...

**Pull request merged**